File: java8/src/test/java/com/shekhargulati/ninetynine_problems/_01_lists/P24Test.java
package com.shekhargulati.ninetynine_problems._01_lists;

import org.junit.Test;

import java.util.List;

import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
import static org.junit.Assert.assertThat;

public class P24Test {

    @Test
    public void shouldGive6RandomNumbersFromARangeStartingFrom1To49() throws Exception {
        List<Integer> randomList = P24.randomSelect(6, 1, 49);
        assertThat(randomList, hasSize(6));
        System.out.println(randomList); // One possible output [47, 30, 36, 38, 11, 1]
    }
}

File: scala/src/test/scala/com/shekhargulati/ninetynine_problems/_01_lists/P24Test.scala
package com.shekhargulati.ninetynine_problems._01_lists

import org.scalatest.{FunSpec, Matchers}

class P24Test extends FunSpec with Matchers {

  describe("P24 Spec") {
    it("should give 6 random number between 1 and 49") {
      val randomList: List[Int] = P24.randomSelect(6, (1, 49))
      randomList should have length 6
      println(randomList) // One possible output List(18, 47, 6, 43, 48, 38)
    }
  }

}